docs/book/driver-deployment/installation.md

```diff
@@ -0,0 +1,226 @@
+# vSphere CSI Driver - Installation
+
+The following steps need to be performed on the k8s node where the vSphere CSI drivers controller will be deployed. We recommend installing vSphere CSI driver on your k8s master node.
+
+## Taint Master Node
+
+Make sure that the node is tainted with `node-role.kubernetes.io/master=:NoSchedule`
+
+```cgo
+$ kubectl describe nodes | egrep "Taints:|Name:"
+Name:               <k8s-master-name>
+Taints:             node-role.kubernetes.io/master:NoSchedule
+Name:               <k8s-worker1-name>
+Taints:             <none>
+Name:               <k8s-worker2-name>
+Taints:             <none>
+Name:               <k8s-worker3-name>
+Taints:             <none>
+Name:               <k8s-worker4-name>
+Taints:             <none>
+```
+
+If it is not tainted, you may do so by running the command:
+
+```cgo
+kubectl taint nodes <k8s-master-name> node-role.kubernetes.io/master=:NoSchedule
+```
+
+## Create a configuration file with vSphere credentials
+
+Create a configuration file that will contain details to connect to vSphere.
+
+The default file to write these configuration details is the `csi-vsphere.conf` file. If you would like to use a file with another name, change the environment variable `VSPHERE_CSI_CONFIG` in the deployment YAMLs described in the section "Install the vSphere CSI driver" below.
+
+### vSphere configuration file for block volumes
+
+Here is an example vSphere configuration file for block volumes, with dummy values:
+
+```cgo
+$ cat /etc/kubernetes/csi-vsphere.conf
+[Global]
+cluster-id = "<cluster-id>"
+
+[VirtualCenter "<IP or FQDN>"]
+insecure-flag = "<true or false>"
+user = "<username>"
+password = "<password>"
+port = "<port>"
+ca-file = <ca file path> # optional, use with insecure-flag set to false
+datacenters = "<datacenter-name1>, <datacenter-name2>, ..."
+```
+
+Where the entries have the following meaning:
+
+- `cluster-id` - represents the unique cluster identifier. Each kubernetes cluster should have it's own unique cluster-id set in the configuration file.
+
+- `VirtualCenter` - section defines vCenter IP address / FQDN.
+
+- `insecure-flag` - should be set to true to use self-signed certificate for login.
+
+- `user` - vCenter username.
+
+- `password` - password for vCenter user specified with user.
+
+- `port` - vCenter Server Port. The default is 443 if not specified.
+
+- `ca-file` - path to a CA certificate in PEM format. It is an optional parameter.
+
+- `datacenters` - list of all comma separated datacenters where kubernetes node VMs are present.
+
+### vSphere configuration file for file volumes
+
+For file volumes, there are some extra parameters added to the config to help with the network permissions and placement of volumes. A sample config file for file volumes is shown below.
```

Comment:
`to the config to help with the network permissions` => `to the config to help specify network permissions`.

Reply:
done